demos/text_detection_demo/cpp_gapi/src/text_recognition.cpp

```diff
@@ -0,0 +1,208 @@
+    }
+    *prob = 1.0f / static_cast<float>(sum);
+}
+
+std::vector<float> softmax(const std::vector<float>::const_iterator& begin,
+                           const std::vector<float>::const_iterator& end) {
+    std::vector<float> prob(end - begin, 0.f);
+    std::transform(begin, end, prob.begin(), [](float x) { return std::exp(x); });
+    float sum = std::accumulate(prob.begin(), prob.end(), 0.0f);
+    for (int i = 0; i < static_cast<int>(prob.size()); i++)
+        prob[i] /= sum;
+    return prob;
+}
+
+struct BeamElement {
+    std::vector<int> sentence;  //!< The sequence of chars that will be a result of the beam element
+    float probBlank;            //!< The probability that the last char in CTC sequence
+                                //!< for the beam element is the special blank char
+    float probNotBlank;         //!< The probability that the last char in CTC sequence
+                                //!< for the beam element is NOT the special blank char
+
+    float prob() const {        //!< The probability of the beam element.
+        return probBlank + probNotBlank;
+    }
+};
+
+std::string SimpleDecoder(const std::vector<float>& data, const std::string& alphabet,
+                          char padSymbol, double *conf, int startIdx) {
+    std::string result = "";
+    const int numClasses = alphabet.length();
+    *conf = 1;
+
+    for (auto it = data.begin() + startIdx * numClasses; it != data.end(); it += numClasses) {
+        int argmax;
+        float prob;
+
+        softmaxAndChoose(it, it + numClasses, &argmax, &prob);
+        (*conf) *= prob;
+        auto symbol = alphabet[argmax];
+        if (symbol != padSymbol)
+            result += symbol;
+        else
+            break;
+    }
+
+    return result;
+}
+
+std::string CTCGreedyDecoder(const std::vector<float>& data, const std::string& alphabet,
+                             char padSymbol, double *conf) {
+    std::string result = "";
+    bool padPrev = false;
+    *conf = 1;
+
+    const int numClasses = alphabet.length();
+    for (auto it = data.begin(); it != data.end(); it += numClasses) {
+      int argmax;
+      float prob;
+
+      softmaxAndChoose(it, it + numClasses, &argmax, &prob);
+
+      (*conf) *= prob;
+
+      auto symbol = alphabet[argmax];
+      if (symbol != padSymbol) {
+          if (result.empty() || padPrev || (!result.empty() && symbol != result.back())) {
+            padPrev = false;
+            result += symbol;
+          }
+      } else {
+        padPrev = true;
+      }
+    }
+
+    return result;
+}
+
+std::string CTCBeamSearchDecoder(const std::vector<float>& data, const std::string& alphabet,
+                                 char padSymbol, double *conf, int bandwidth) {
+    const int numClasses = alphabet.length();
+
+    std::vector<BeamElement> curr;
+    std::vector<BeamElement> last;
+
+    last.push_back(BeamElement{std::vector<int>(), 1.f, 0.f});
```

Comment:
MB `emplace_back(std::vector<int>(), 1.f, 0.f)`

Reply:
NO, `BeamElement` struct doesn't have a constructor so `emplace` doesn't work - and I don't want to change a lot in these functions, let's leave them as similar with `cpp` as it is reasonable